### PR TITLE
Change to Font Awesome icon font (fixes #2138)

### LINKED ISSUE #2138: Change to fontawesome icon font
I suggest to switch to the [fontawesome](http://fontawesome.io) icon font. Besides having a lot more [icons](http://fontawesome.io/icons/) to choose from for future functions in Syncthing it has many every day symbols like desktop, mobile, tablet (which could be used as a user setting instead of the identicon), twitter, wifi, etc. The transition should be fairly quick as most icons have a similar look and same name.

### PR DESCRIPTION
**Pull request by uok**
- add vendor assets for Font Awesome v4.4.0
- replace Glyphicons with Font Awesome icons
- add icons to modal headers

### CODE REVIEW

**Comment by AudriusButkevicius** [comment]
So haven't fully gone through it, but I don't like cramming stuff up into a single line unlike it was expanded before in some places

**Comment by ralder** [comment]
also may add ``` .fa {margin-right: 7px; } ``` and remove corresponding ```&nbsp;``` from html

File: gui/syncthing/usagereport/usageReportPreviewModalView.html [hunk 1]
```diff
@@ -1,17 +1,5 @@
...
-	      <button type="button" class="btn btn-success btn-sm" data-dismiss="modal"><span class="glyphicon glyphicon-ok"></span>&nbsp;<span translate>OK</span></button>
-	    </div>
-	  </div>
-	</div>
-</div>
+<modal id="urPreview" status="success" icon="bar-chart" title="{{'Anonymous Usage Reporting' | translate}}" large="yes" close="yes" tabindex="-1">
+  <p translate>The encrypted usage report is sent daily. It is used to track common platforms, folder sizes and app versions. If the reported data set is changed you will be prompted with this dialog again.</p>
+  <p translate translate-value-url="<a href=&quot;https://data.syncthing.net&quot; target=&quot;_blank&quot;>https://data.syncthing.net</a>">The aggregated statistics are publicly available at {%url%}.</p>
+  <pre><small>{{reportData | json}}</small></pre>
+</modal>
```
Comment (canton7) [inline gui/syncthing/usagereport/usageReportPreviewModalView.html]: This seems to be quite different?

File: gui/syncthing/folder/editFolderModalView.html [hunk 2]
```diff
@@ -2,8 +2,10 @@
   <div class="modal-dialog modal-lg">
     <div class="modal-content">
       <div class="modal-header">
-        <h4 ng-show="!editingExisting" class="modal-title"><span translate>Add Folder</span></h4>
-        <h4 ng-show="editingExisting" class="modal-title"><span translate>Edit Folder</span></h4>
+        <h4 class="modal-title">
+          <span ng-show="!editingExisting"><span class="fa fa-folder"></span><span translate>Add Folder</span></span>
+          <span ng-show="editingExisting"><span class="fa fa-pencil"></span><span translate>Edit Folder</span></span>
```
Comment (canton7) [inline gui/syncthing/folder/editFolderModalView.html]: Semantic change? Two `h4`s into one `h4`

**Comment by canton7** [comment]
The attributation / third party components stuff probably wants updating as well.

File: gui/syncthing/folder/editFolderModalView.html (hunk 2 above)
Comment (AudriusButkevicius) [inline gui/syncthing/folder/editFolderModalView.html]: Its conditional, so no.

**Comment by calmh** [comment]
Doesn't actually remove Glyphicons; do we need to ship both...?

**Comment by uok** [comment]
@AudriusButkevicius do you mean the `<button>` tag? Since I found both variants in the source code I chose the one that was used more often (and IMHO better suited for this tag). If you don't like the HTML formatting please provide a simple windows tools that does that as I only use a simple text editor and don't want to spend my free time manually doing what could be done automatically. Also this should be done in a separate PR since most of the HTML code has not been touched for many months.

@ralder my PR is just for the icons, not HTML. I just copied it from all the other buttons. Also AFAIK it is better to use `nbsp;` here because it prevents a line break inside the button. If you are 100% sure this works with padding as well please open another PR

@canton7 please point me to the files that need this info

@calmh you are probably right, the font files could be removed (.eot, .svg, .ttf, etc.). But the icons are also defined inside the bootstrap CSS - to get them out of there you need to [compile](http://getbootstrap.com/customize/) your own bootstrap files. The you need to make sure that the next person that updates bootstrap in a few months remembers that

File: gui/syncthing/usagereport/usageReportPreviewModalView.html (hunk 1 above)
Comment (uok) [inline gui/syncthing/usagereport/usageReportPreviewModalView.html]: whoever extraced the modals into separate files forgot about this one, so I corrected that. It is the same content but it uses the angular directive which then loads the above into modal.html which adds a close button at the bottom and the modal header - voila!

**Comment by calmh** [comment]
@uok No worries on that point. Our bootstrap is custom anyway, with the settings used in `gui/vendor/bootstrap/config.json`. That file can be dropped in the download section on getbootstrap.com for our current settings, the fints stuff removed (I think) and a new version downloaded.

**Comment by uok** [comment]
this worked perfectly, the new config.json and bootstrap files are all without glyphicons now

**Comment by calmh** [comment]
@st-jenkins test this please

File: gui/syncthing/transfer/neededFilesModalView.html [hunk 3]
```diff
@@ -1,4 +1,4 @@
-<div id="needed" class="modal fade" large="yes" status="info" icon="cloud-download" close="yes" title="{{'Out of Sync Items' | translate}}" tabindex="-1">
+<modal id="needed" status="info" icon="cloud-download" title="{{'Out of Sync Items' | translate}}" large="yes" close="yes" tabindex="-1">
```
Comment (calmh) [inline gui/syncthing/transfer/neededFilesModalView.html]: :+1:

**Comment by calmh** [comment]
I agree with Audrius about the button changes - the rule of thumb being keep it on one line, unless that line gets too long. :) It's not a big deal though. Can you rebase this on your other changes (it doesn't merge currently)?

**Comment by uok** [comment]
as recommended I formatted the HTML source for all `<button>` elements.
rebase done!

**Comment by calmh** [comment]
Hmm, you seem rebased on top of v0.11.19 or so, not the current `master`. Do you need a pull first?

**Comment by uok** [comment]
Is it possible that when I started master = 0.11.19 but in the meantime something changed? I always

    git checkout master
    git fetch upstream
    git merge upstream/master

before I create a new branch. So far this has always worked, I don't know any other commands so far but I'm happy to learn (and put them on my "github instructions" list)

**Pull request merged**